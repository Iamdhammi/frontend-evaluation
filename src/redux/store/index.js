import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState = {};

const middleware = [thunk];

// create Store
const store = createStore(
    rootReducer, // all combined reducers
    initialState, // DUHH initaial state at craete store level
    compose(
        applyMiddleware(...middleware)
    )
);

export default store;