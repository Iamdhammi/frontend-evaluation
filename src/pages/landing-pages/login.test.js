import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Login from './login';
import { Router } from "react-router-dom";
import history from '../../history'

afterEach(cleanup);

it("loads Login component", () => {
    const { asFragment } = render(
        <Provider store={store}>
            <Router history={history}>
                <Login />
            </Router>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

describe("login integration", () => {
  it("User should be able to login", async () => {
    const promise = Promise.resolve();
    const { getByText, getByPlaceholderText } = render(
        <Provider store={store}>
            <Router history={history}>
                <Login />
            </Router>
        </Provider>
    );

    userEvent.type(getByPlaceholderText("Email"), "test@test.com" );
    userEvent.type(getByPlaceholderText("Password"), "test@test");

    const SubmitBtn = getByText(/Log in/i);
    userEvent.click(SubmitBtn);

    expect(getByPlaceholderText("Email")).toHaveAttribute(
      "value",
      "test@test.com"
    );

    expect(getByPlaceholderText("Password")).toHaveAttribute(
      "value",
      "test@test"
    );

    await act(() => promise);
  });
});