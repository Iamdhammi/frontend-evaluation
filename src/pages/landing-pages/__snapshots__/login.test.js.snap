// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads Login component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="login-container"
    >
      <div
        class="grid grid--cols-2"
      >
        <div
          class="grid-cell-1-5"
        >
          <div
            class="login-container__content"
          >
            <div
              class="content"
            >
              <div
                class="content__logo"
              >
                <img
                  alt="Logo"
                  class="content__logo--img"
                  src="logo.png"
                />
              </div>
              <div
                class="content__text"
              >
                <p>
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos soluta fugit, fugiat necessitatibus aperiam minima molestias sint enim ipsum perferendis maxime culpa similique modi tempora, aliquid quasi magnam numquam doloremque.
                </p>
                <p>
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos soluta fugit, fugiat necessitatibus aperiam minima molestias sint enim ipsum perferendis maxime culpa similique modi tempora, aliquid quasi magnam numquam doloremque.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid-cell"
        >
          <div
            class="login-container__fields"
          >
            <div
              class="Toastify"
            />
            <div
              class="content"
            >
              <div
                class="content__header"
              >
                <h2>
                  Proceed to Archimydes
                </h2>
              </div>
              <form
                class="ui form"
              >
                <div
                  class="field"
                >
                  <div
                    class="form__control"
                  >
                    <label>
                      Email Address
                    </label>
                    <div
                      class="ui left icon input form__control--input"
                    >
                      <i
                        aria-hidden="true"
                        class="mail icon"
                      />
                      <input
                        placeholder="Email"
                        type="email"
                        value=""
                      />
                    </div>
                    <span
                      class="span-error"
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="form__control"
                  >
                    <label>
                      Password
                    </label>
                    <div
                      class="ui left icon input form__control--input"
                    >
                      <i
                        aria-hidden="true"
                        class="key icon"
                      />
                      <input
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <span
                      class="span-error"
                    />
                    <button
                      class="ui icon button form__control--icon"
                    >
                      <i
                        aria-hidden="true"
                        class="eye icon"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="form__control"
                  >
                    <div
                      class="ui checkbox"
                    >
                      <input
                        class="hidden"
                        readonly=""
                        tabindex="0"
                        type="checkbox"
                        value=""
                      />
                      <label>
                        Proceed as an Admin
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="form__control"
                >
                  <button
                    class="ui fluid button form__control--button"
                    type="submit"
                  >
                    <i
                      aria-hidden="true"
                      class="mail icon"
                    />
                    Log in
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
